src/process_utility.c

```diff
@@ -233,11 +234,64 @@ check_alter_table_allowed_on_ht_with_compression(Hypertable *ht, AlterTableStmt
 			case AT_AddColumn:	 /* this is passed down */
 			case AT_ColumnDefault: /* this is passed down */
 			case AT_DropColumn:	/* this is passed down */
+
+			case AT_DropConstraint:
+			case AT_AlterConstraint:
 #if PG14_GE
 			case AT_ReAddStatistics:
 			case AT_SetCompression:
 #endif
 				continue;
+
+			/*
+			 * Conditionally allowed command. Adding a unique constraint
+			 * including non-segmentby column is not supported.
+			 */
+			case AT_AddConstraint:
+			{
+				Constraint *constraint = (Constraint *) cmd->def;
+
+				if (constraint->contype == CONSTR_UNIQUE)
+				{
+					List *allowed = NIL, *info = NIL;
+					ListCell *lc;
+					int i;
+
+					info = ts_hypertable_compression_get(ht->fd.id);
+
+					/* Unique constraints on segmentby columns are supported */
+					foreach (lc, info)
+					{
+						FormData_hypertable_compression *fd = lfirst(lc);
+						if (fd->segmentby_column_index > 0)
+							allowed = lappend(allowed,
+											  makeString(NameStr(fd->attname)));
```

Comment:
Codecov complains about this line not being covered (you should see the annotation below). Could you add a test for this?